### PR TITLE
#31 Content type and length switching rules

### PR DESCRIPTION
**Pull request by jakeasmith**
Provides a patch to work around a bug in the built in PHP server.

### CODE REVIEW

File: src/Request/Content.php [hunk 1]
```diff
@@ -101,6 +101,15 @@ public function __construct(
                    ? strtolower($server['HTTP_CONTENT_MD5'])
                    : null;
 
+	    // Catches the content values with "HTTP_" prefix. This addresses a bug 
+	    // in the built in PHP server https://bugs.php.net/bug.php?id=66606
+	    if (empty($this->type) && !empty($_SERVER['HTTP_CONTENT_TYPE'])) {
+		    $this->type = $_SERVER['HTTP_CONTENT_TYPE'];
```
Comment (harikt) [inline src/Request/Content.php:107]: the spacing seems to be a tab to me. Yes I am looking at https://github.com/jakeasmith/Aura.Web/blob/6f8dca3beb82f5face930f07d9361ff69db90b7b/src/Request/Content.php#L104-L111 .

**Comment by harikt** [comment]
Paul, we have discussed this earlier on https://github.com/auraphp/Aura.Web/pull/24#issuecomment-33841856 . May be you have changed your mind?

**Comment by pmjones** [comment]
@jakeasmith see #23 and #24 for some background. I am loathe to add behavior that "fixes" for the PHP internal server, but this is the second time it's come up, so I guess it really is a problem.

This PR uses the $_SERVER super global instead of the injected $server, which is not the approved way of doing things. ;-) It also needs to deal with CONTENT_LENGTH (although strangely not CONTENT_MD5).  And then there's the tabs. ;-)

In all, I'm going to go back to @garoevans #24 , merge it, and modify it somewhat. We'll see if that does the trick.

Thanks @jakeasmith for both the report and the PR; if nothing else it proves this really is a problem,

**Comment by jakeasmith** [comment]
Oh wow, that was a mess. Thanks for picking this up though, I appreciate it.

**Comment by pmjones** [comment]
Well, @garoevans removed his issue branch, so I can't merge that one. So we're back to you!

Do me a favor: rewrite this to use $server, check both content-type and content-length, replace the tabs, and we'll go from there. Take a look at @garoevans getHeaderValue() method while you're at it to make things a little cleaner.

Thanks for your patience dude.

**Comment by jakeasmith** [comment]
No worries. I'll clean it up this evening and kick it back to you.

**Comment by harikt** [comment]
@pmjones It was not the PR from @garoevans . It was the PR from me.

I have restored the branch.

Have a look  https://github.com/auraphp/Aura.Web/pull/24 

Thank you

**Comment by garoevans** [comment]
I have changes in my local projects that deal with this, but never PR'd,
just bug report and conversation. Sounds like you guys have this all
covered, but am happy to double check the PR for my use case as well if
that helps?

On 27 July 2014 03:10, Hari K T notifications@github.com wrote:

> @pmjones https://github.com/pmjones It was not the PR from @garoevans
> https://github.com/garoevans . It was the PR from me.
> 
> I have restored the branch.
> 
> Have a look #24 https://github.com/auraphp/Aura.Web/pull/24
> 
> Thank you
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/auraphp/Aura.Web/pull/32#issuecomment-50253614.

## 

t: @garoevans
l: http://uk.linkedin.com/in/garoevans/
m: +44 7718 617228

**Comment by garoevans** [comment]
Also, I dealt with it in define.php before the super globals became local
`$server` variables, so not jumping into global scope inside the
application.

On 28 July 2014 08:38, Gareth Evans garoevans@gmail.com wrote:

> I have changes in my local projects that deal with this, but never PR'd,
> just bug report and conversation. Sounds like you guys have this all
> covered, but am happy to double check the PR for my use case as well if
> that helps?
> 
> On 27 July 2014 03:10, Hari K T notifications@github.com wrote:
> 
> > @pmjones https://github.com/pmjones It was not the PR from @garoevans
> > https://github.com/garoevans . It was the PR from me.
> > 
> > I have restored the branch.
> > 
> > Have a look #24 https://github.com/auraphp/Aura.Web/pull/24
> > 
> > Thank you
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/auraphp/Aura.Web/pull/32#issuecomment-50253614.
> 
> ## 
> 
> t: @garoevans
> l: http://uk.linkedin.com/in/garoevans/
> m: +44 7718 617228

## 

t: @garoevans
l: http://uk.linkedin.com/in/garoevans/
m: +44 7718 617228

**Pull request merged**